src/jit/codegenxarch.cpp

```diff
@@ -6192,18 +6256,30 @@ void CodeGen::genCallInstruction(GenTreePtr node)
             {
                 assert((arg->gtGetOp1()->OperGet() == GT_PUTARG_STK) && (arg->gtGetOp2()->OperGet() == GT_PUTARG_STK));
             }
+            if ((arg->OperGet() == GT_PUTARG_STK) && (arg->gtGetOp1()->OperGet() == GT_FIELD_LIST))
```

Comment:
Actually, this can't be an `else if` because in the long case we need to execute the else condition below.

---
In reply to: [79508507](https://github.com/dotnet/coreclr/pull/7252#discussion_r79508507) [](ancestors = 79508507)